### PR TITLE
Fix DNS Record resource

### LINKED ISSUE #62: DNS record errors
Running this terraform code:
```Terraform
terraform {
  required_version = ">= 1.2.7"

  required_providers {
    digitalocean = {
      source  = "digitalocean/digitalocean"
      version = ">= 2.22.1"
    }
    vercel = {
      source  = "vercel/vercel"
      version = ">= 0.7.1"
    }
  }

  cloud {
    organization = "syncbase"

    workspaces {
      name = "syncbase"
    }
  }

resource "vercel_project" "frontend" {
  name      = "frontend-syncbase"
  framework = "nextjs"
}

resource "vercel_project_domain" "default" {
  project_id = vercel_project.frontend.id
  domain     = "syncbase.tv"
}

resource "vercel_dns_record" "a" {
  domain = vercel_project_domain.default.domain
  name   = ""
  type   = "A"
}

resource "vercel_dns_record" "cname" {
  domain = vercel_project_domain.default.domain
  name   = "www"
  type   = "CNAME"
  value  = vercel_project_domain.default.domain
}
```
Raises the following two errors:
```
╷
│ Error: Provider produced inconsistent result after apply
│
│ When applying changes to vercel_dns_record.cname, provider "provider[\"registry.terraform.io/vercel/vercel\"]" produced an unexpected new value: .ttl: was null, but
│ now cty.NumberIntVal(60).
│
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
╵
╷
│ Error: Error creating DNS Record
│
│   with vercel_dns_record.a,
│   on main.tf line 103, in resource "vercel_dns_record" "a":
│  103: resource "vercel_dns_record" "a" {
│
│ Could not create DNS Record, unexpected error: bad_request - Invalid request: missing required property `value`.
╵
```
As stated in the docs:
image
Both the `ttl` and `value` option are optional so they shouldn't need be specified in the terraform configuration

### PR DESCRIPTION
**Pull request by ecklf**
This PR has the following changes:

- Fixes a bug where omitting `ttl` (defaults to 60) from the resource creates an inconsistent result.
- Adds tests for the above edge case.
- Adds additional validation for preventing to omit `value` for all DNS Record types except of type `SRV`

Closes #62

### CODE REVIEW

**Review by dglsparsons — DISMISSED** [review]

File: vercel/resource_dns_record_test.go [hunk 1]
```diff
@@ -137,6 +143,11 @@ func TestAcc_DNSRecord(t *testing.T) {
 			{
 				Config: testAccDNSRecordConfigUpdated(testDomain(), nameSuffix),
 				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccDNSRecordExists("vercel_dns_record.a_without_ttl", ""),
+					resource.TestCheckResourceAttr("vercel_dns_record.a", "domain", testDomain()),
+					resource.TestCheckResourceAttr("vercel_dns_record.a", "type", "A"),
+					resource.TestCheckResourceAttr("vercel_dns_record.a", "ttl", "120"),
+					resource.TestCheckResourceAttr("vercel_dns_record.a", "value", "127.0.0.1"),
```
Comment (dglsparsons) [inline vercel/resource_dns_record_test.go:150]:
```suggestion
					resource.TestCheckResourceAttr("vercel_dns_record.a_without_ttl", "domain", testDomain()),
					resource.TestCheckResourceAttr("vercel_dns_record.a_without_ttl", "type", "A"),
					resource.TestCheckResourceAttr("vercel_dns_record.a_without_ttl", "ttl", "120"),
					resource.TestCheckResourceAttr("vercel_dns_record.a_without_ttl", "value", "127.0.0.1"),
```

File: vercel/resource_dns_record_test.go [hunk 2]
```diff
@@ -76,6 +77,11 @@ func TestAcc_DNSRecord(t *testing.T) {
 			{
 				Config: testAccDNSRecordConfig(testDomain(), nameSuffix),
 				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccDNSRecordExists("vercel_dns_record.a_without_ttl", ""),
+					resource.TestCheckResourceAttr("vercel_dns_record.a", "domain", testDomain()),
+					resource.TestCheckResourceAttr("vercel_dns_record.a", "type", "A"),
+					resource.TestCheckResourceAttr("vercel_dns_record.a", "ttl", "60"),
+					resource.TestCheckResourceAttr("vercel_dns_record.a", "value", "127.0.0.1"),
```
Comment (dglsparsons) [inline vercel/resource_dns_record_test.go:84]:
```suggestion
					resource.TestCheckResourceAttr("vercel_dns_record.a_without_ttl", "domain", testDomain()),
					resource.TestCheckResourceAttr("vercel_dns_record.a_without_ttl", "type", "A"),
					resource.TestCheckResourceAttr("vercel_dns_record.a_without_ttl", "ttl", "60"),
					resource.TestCheckResourceAttr("vercel_dns_record.a_without_ttl", "value", "127.0.0.1"),
```

**Review by dglsparsons — APPROVED** [review]
